multiplier_analysis/nbconverted/1_get_multiplier_LV_coverage.py

```diff
@@ -0,0 +1,249 @@
+# 
+# While the PLIER model was trained on specific datasets, MULTIPLIER extended this approach to all of recount2, where the latent variables should correspond to specific pathways or gene sets of interest. Therefore, we will look at the coverage of generic genes versus specific genes across these MULTIPLIER latent variables.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import pandas as pd
+import seaborn as sns
+
+import rpy2.robjects as ro
+from rpy2.robjects import pandas2ri
+from rpy2.robjects.conversion import localconverter
+
+from ponyo import utils
+from generic_expression_patterns_modules import process
+
+
+# In[2]:
+
+
+# Get data directory containing gene summary data
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+data_dir = os.path.join(base_dir, "human_general_analysis")
+
+# Read in config variables
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_human_general.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+local_dir = params["local_dir"]
+
+
+# In[3]:
+
+
+# Output file
+output_figure_filename = "LV_coverage.svg"
+
+
+# ## Get gene data
+
+# In[4]:
+
+
+# Get all gene summary file paths
+ls_data_files = process.get_gene_summary_files(data_dir)
+
+
+# In[5]:
+
+
+# Identify generic and specific genes using z-score
+# Z-score cutoff was found by calculating invnorm(0.05/17754). 
+# To do this in python you can use the following code:
+# from scipy.stats import norm
+# norm.ppf((0.05/17754)/2)
+# Here we are using a p-value = 0.05
+# with a Bonferroni correction for 17754 tests, which are
+# the number of P. aeruginosa genes
+
+zscore_threshold = 4.68
+ls_genes_out = process.get_generic_specific_genes(ls_data_files, zscore_threshold)
+
+
+# ## Get LV data and filter
+
+# In[6]:
+
+
+# Load multiplier models
+# Converted formatted pickle files (loaded using phenoplier environment) from
+# https://github.com/greenelab/phenoplier/blob/master/nbs/01_preprocessing/005-multiplier_recount2_models.ipynb
+# into .tsv files
+# Raw data was downloaded from https://figshare.com/articles/recount_rpkm_RData/5716033/4
+multiplier_model_summary = pd.read_csv("multiplier_model_summary.tsv", sep="\t", index_col=0, header=0)
+multiplier_model_z = pd.read_csv("multiplier_model_z.tsv", sep="\t", index_col=0, header=0)
+
+
+# In[7]:
+
+
+multiplier_model_summary.head()
+
+
+# In[8]:
+
+
+# Only select LVs that are signficantly associated with some pathways or gene set (i.e. FDR < 0.05)
+multiplier_model_z_processed = process.process_multiplier_model_z(multiplier_model_z, multiplier_model_summary)
+
+
+# In[9]:
+
+
+multiplier_model_z_processed.head()
+
+
+# ## Quick looks at the data
+
+# In[10]:
+
+
+# Get a rough sense for how many genes contribute to a given LV
+# (i.e. how many genes have a value > 0 for a given LV)
+(multiplier_model_z > 0).sum()
```

Comment:
Maybe you could sort this, to get a sense of what the bounds are. Are there any LVs that are associated with only a few genes?

Reply:
Good suggestion. This was definitely a check to see if there were any LVs with 0 genes contributing, which there doesn't look to be, and to possibly filter those out